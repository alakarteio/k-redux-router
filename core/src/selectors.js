export default (getState) => {
  const getRoute = code => state => getState(state).routes.map[code]
  const getResult = state => getState(state).result
  const getCurrentRoute = state => getResult(state).route
  const getCurrentCode = state => getCurrentRoute(state).code
  const isFound = state => getResult(state).found
  const getParams = state => getResult(state).params
  const getPathParams = state => getParams(state).path
  const getQueryParams = state => getParams(state).query
  const getPathParam = code => state => getPathParams(state)[code]
  const getQueryParam = code => state => getQueryParams(state)[code]
  const getParam = code => state => getPathParam(code)(state) || getQueryParam(code)(state)

  return {
    getState,
    getRoute,
    getResult,
    getCurrentCode,
    getCurrentRoute,
    isFound,
    getParams,
    getPathParams,
    getQueryParams,
    getPathParam,
    getQueryParam,
    getParam,
  }
}
