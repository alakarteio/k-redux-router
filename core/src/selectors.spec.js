/* eslint-env jest */
import selectors from './selectors'

describe('selectors', () => {
  const getState = () => ({
    otherPath: {
      result: {
        found: true,
        route: {
          code: 'other-path',
        },
      },
    },
    notFound: {
      result: {
        found: false,
      },
    },
    router: {
      routes: {
        map: {
          socials: {
            code: 'socials',
            custom: 'custom-socials-value',
          },
        },
      },
      result: {
        found: true,
        route: {
          code: 'current-code',
          custom: 'custom-param-value',
        },
        params: {
          path: {
            code: 'path-value',
            other: 'path-other-value',
          },
          query: {
            code: 'query-value',
            next: 'query-next-value',
          },
        },
      },
    },
  })

  describe('factory', () => {
    const match = callback => () => expect(callback(getState())).toMatchSnapshot()

    // test selector
    it('should select to otherPath', match(selectors(state => state.otherPath).getCurrentCode))

    // all the next tests are from the same selector
    const s = selectors(state => state.router)
    it('should get route code (current-code)', match(s.getCurrentCode))
    it('should get param from path (path-value)', match(s.getParam('code')))
    it('should get param from query (query-next-value)', match(s.getParam('next')))
    it('should get query param (query-next-value)', match(s.getQueryParam('next')))
    it('should get path param (path-value)', match(s.getPathParam('code')))
    it('should get all the query params', match(s.getQueryParams))
    it('should get all the path params', match(s.getPathParams))
    it('should get all the params', match(s.getParams))
    it('should found the route', match(s.isFound))
    it('should get the current route definition', match(s.getCurrentRoute))
    it('should get the result', match(s.getResult))
    it('should get the route -socials-', match(s.getRoute('socials')))
  })
})
