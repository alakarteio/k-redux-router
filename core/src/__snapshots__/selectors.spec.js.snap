// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors factory should found the route 1`] = `true`;

exports[`selectors factory should get all the params 1`] = `
Object {
  "path": Object {
    "code": "path-value",
    "other": "path-other-value",
  },
  "query": Object {
    "code": "query-value",
    "next": "query-next-value",
  },
}
`;

exports[`selectors factory should get all the path params 1`] = `
Object {
  "code": "path-value",
  "other": "path-other-value",
}
`;

exports[`selectors factory should get all the query params 1`] = `
Object {
  "code": "query-value",
  "next": "query-next-value",
}
`;

exports[`selectors factory should get param from path (path-value) 1`] = `"path-value"`;

exports[`selectors factory should get param from query (query-next-value) 1`] = `"query-next-value"`;

exports[`selectors factory should get path param (path-value) 1`] = `"path-value"`;

exports[`selectors factory should get query param (query-next-value) 1`] = `"query-next-value"`;

exports[`selectors factory should get route code (current-code) 1`] = `"current-code"`;

exports[`selectors factory should get the current route definition 1`] = `
Object {
  "code": "current-code",
  "custom": "custom-param-value",
}
`;

exports[`selectors factory should get the result 1`] = `
Object {
  "found": true,
  "params": Object {
    "path": Object {
      "code": "path-value",
      "other": "path-other-value",
    },
    "query": Object {
      "code": "query-value",
      "next": "query-next-value",
    },
  },
  "route": Object {
    "code": "current-code",
    "custom": "custom-param-value",
  },
}
`;

exports[`selectors factory should get the route -socials- 1`] = `
Object {
  "code": "socials",
  "custom": "custom-socials-value",
}
`;

exports[`selectors factory should select to otherPath 1`] = `"other-path"`;
